python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/ethernet-interfaces.j2

```diff
@@ -45,19 +45,41 @@
 {%             endif %}
 {%         elif ethernet_interface.type | arista.avd.default('switched') == 'switched' %}
 {%             set description = ethernet_interface.description | arista.avd.default("-") %}
-{%             set mode = ethernet_interface.mode | arista.avd.default("access") %}
-{%             set vlans = ethernet_interface.vlans | arista.avd.default('-') %}
-{%             if  ethernet_interface.native_vlan_tag is arista.avd.defined(true) %}
+{%             if  ethernet_interface.switchport.mode is arista.avd.defined %}
+{%                 set mode = ethernet_interface.switchport.mode %}
+{%                 if mode in ["access", "dot1q-tunnel"] %}
+{%                     set vlans = ethernet_interface.switchport.access_vlan %}
+{%                 elif mode == "trunk" %}
+{%                     set vlans = ethernet_interface.switchport.trunk.allowed_vlan %}
+{%                 elif mode == "trunk phone" %}
+{%                     set vlans = ethernet_interface.switchport.phone.vlan %}
+{%                 endif %}
+{%             else %}
+{%                 set mode = ethernet_interface.mode | arista.avd.default('access') %}
+{%                 set vlans = ethernet_interface.vlans | arista.avd.default('-') %}
+{%             endif %}
+{%             if  ethernet_interface.native_vlan_tag is arista.avd.defined(true) or
+                ethernet_interface.switchport.trunk.native_vlan_tag is arista.avd.defined(true) %}
 {%                 set native_vlan = "tag" %}
 {%             else %}
-{%                 set native_vlan = ethernet_interface.native_vlan | arista.avd.default('-') %}
+{%                 if ethernet_interface.switchport.trunk.native_vlan is arista.avd.defined %}
+{%                     set native_vlan = ethernet_interface.switchport.trunk.native_vlan %}
+{%                 else %}
+{%                     set native_vlan = ethernet_interface.native_vlan | arista.avd.default('-') %}
+{%                 endif %}
 {%             endif %}
-{%             if ethernet_interface.trunk_groups is defined %}
+{%             if ethernet_interface.trunk_groups is arista.avd.defined or ethernet_interface.switchport.trunk.groups is arista.avd.defined %}
```

Comment:
I think everything within this block can be simplified a great deal by using the default filter to get either list (new or old) or fall back to `['-']`

Reply:
Updated!